app/src/main/java/com/hedvig/app/feature/offer/usecase/getquote/GetQuotesUseCase.kt

```diff
@@ -17,34 +19,18 @@ class GetQuotesUseCase(
     private val featureManager: FeatureManager,
     private val getQuoteIdsUseCase: GetQuoteIdsUseCase,
 ) {
-    sealed class Result {
-        data class Success(val data: OfferModel) : Result()
-        data class Error(val message: String?) : Result()
-    }
 
-    operator fun invoke(quoteIds: List<String>, quoteCartId: String?): Flow<Result> {
+    operator fun invoke(quoteIds: List<String>, quoteCartId: String?): Flow<Either<ErrorMessage, OfferModel>> {
         return flow {
             when {
-                featureManager.isFeatureEnabled(Feature.QUOTE_CART) -> {
-                    quoteCartId
-                        ?.let { emitAll(getOffer(nonEmptyListOf(quoteCartId))) }
-                        ?: emit(Result.Error("No quote cart id found"))
-                }
-                quoteIds.isNotEmpty() -> emitAll(getOffer(NonEmptyList.fromListUnsafe(quoteIds)))
-                else -> {
-                    when (val result = getQuoteIdsUseCase.invoke(null)) {
-                        is Either.Left -> emit(Result.Error(result.value.message))
-                        is Either.Right -> emitAll(getOffer(NonEmptyList.fromListUnsafe(result.value.ids)))
-                    }
+                featureManager.isFeatureEnabled(Feature.QUOTE_CART) -> quoteCartId?.let {
+                    emitAll(offerRepository.offer(nonEmptyListOf(quoteCartId)))
+                } ?: ErrorMessage("No quote id found").left()
```

Comment:
Maybe keep the braces here? Maybe it's just GitHub but it's quite hard to read imo